crates/oxc_linter/src/rules/promise/no_new_statics.rs

```diff
@@ -0,0 +1,106 @@
+use oxc_ast::{ast::Expression, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn static_promise_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(
+        "eslint-plugin-promise(no-new-statics): Disallow calling `new` on a Promise static method",
```

Comment:
Nit: Seems the error message about this rule is `Avoid calling 'new' on 'Promise.{{ name }}()`, let's keep the same with it.

https://github.com/eslint-community/eslint-plugin-promise/blob/ecbce9faef0c8e05636287cf9c851949be6d70cc/rules/no-new-statics.js#L26